defold-rive/pluginsrc/plugin.cpp

```diff
@@ -658,167 +670,128 @@ static void GenerateAABB(RiveFile* file)
+    static const dmhash_t UNIFORM_COLOR = dmHashString64("color");
+    static const dmhash_t UNIFORM_COVER = dmHashString64("cover");
 
-    rive::newFrame(file->m_Renderer);
+    RiveEventsDrawcallContext* plugin_ctx = (RiveEventsDrawcallContext*)ctx->m_UserContext;
 
-    rive::Mat2D transform;
-    rive::Mat2D::identity(transform);
+    switch(ctx->m_Event.m_Type)
+    {
+    case rive::EVENT_DRAW_STENCIL:
+        {
+            dmRive::RenderObject& ro = plugin_ctx->m_RenderObjects[ctx->m_Index];
+            ro.Init();
 
-    rive::Vec2D yflip(1.0f,-1.0f);
-    rive::Mat2D::scale(transform, transform, yflip);
-    rive::setTransform(file->m_Renderer, transform);
+            ro.m_VertexStart       = ctx->m_IndexOffsetBytes; // byte offset
+            ro.m_VertexCount       = ctx->m_IndexCount;
+            ro.m_SetStencilTest    = 1;
+            ro.m_SetFaceWinding    = 1;
+            ro.m_FaceWindingCCW    = ctx->m_FaceWinding == dmGraphics::FACE_WINDING_CCW;
 
-    rive::Artboard* artboard = file->m_File->artboard();
-    rive::AABB bounds = artboard->bounds();
-    float cx = (bounds.maxX - bounds.minX) * 0.5f;
-    float cy = (bounds.maxY - bounds.minY) * 0.5f;
+            dmRive::SetStencilDrawState(&ro.m_StencilTestParams, ctx->m_Event.m_IsClipping, ctx->m_ClearClippingFlag);
 
-    rive::HContext rive_ctx = rive::g_Ctx;
-    rive::HRenderer renderer = file->m_Renderer;
-    rive::Renderer* rive_renderer = (rive::Renderer*) renderer;
+            ro.AddConstant(UNIFORM_COVER, dmVMath::Vector4(0.0f, 0.0f, 0.0f, 0.0f));
 
-    rive_renderer->align(rive::Fit::none, rive::Alignment::center, bounds, bounds);
-    rive_renderer->save();
+            dmRive::Mat2DToMat4(ctx->m_Event.m_TransformWorld, ro.m_WorldTransform);
+        }
+        break;
 
-    // Triggers the buffer allocation callbacks
-    artboard->draw(rive_renderer);
-    rive_renderer->restore();
+    case rive::EVENT_DRAW_COVER:
+        {
+            dmRive::RenderObject& ro = plugin_ctx->m_RenderObjects[ctx->m_Index];
+            ro.Init();
+            ro.m_VertexStart       = ctx->m_IndexOffsetBytes; // byte offset
+            ro.m_VertexCount       = ctx->m_IndexCount;
+            ro.m_SetStencilTest    = 1;
 
-    // Now we also need to iterate over the paint events
+            ro.m_SetFaceWinding    = 1;
+            ro.m_FaceWindingCCW    = ctx->m_FaceWinding == dmGraphics::FACE_WINDING_CCW;
 
-    rive::HRenderPaint paint = 0;
-    bool is_paint_dirty = false;
-    bool is_applying_clipping = false;
-    bool is_clipping = false;
+            dmRive::SetStencilCoverState(&ro.m_StencilTestParams, ctx->m_Event.m_IsClipping, ctx->m_IsApplyingClipping);
 
-    float z = 0.0f;
+            if (!ctx->m_IsApplyingClipping)
+            {
+                // GetBlendFactorsFromBlendMode(plugin_ctx->m_BlendMode, &ro.m_SourceBlendFactor, &ro.m_DestinationBlendFactor);
+                // ro.m_SetBlendFactors = 1;
```

Comment:
Not needed? I dont remember if we did use this

Reply:
Not really sure if the user will set this.
I left it out for this mvp